### DISCUSSION TITLE
run-docker.sh gets error if home folder is used

### DISCUSSION THREAD

**Discussion by mustafatasci42** [Xilinx/finn#673] [Q&A] [answered]
If I install fin in home directory, run-docker.sh gives error but if I install in Opt folder it works fine with sudo command.
![home_folder](https://user-images.githubusercontent.com/15524481/186147903-aa34ce16-8d98-4362-b4ed-fcf518d74615.png)

![opt_folder](https://user-images.githubusercontent.com/15524481/186147861-a8e3cb31-9375-4e7b-82dd-acd6b8f0dd23.png)
I don't want to use sudo what should I do

**Comment by julienmas** [comment]
Hi Mustafa,

Can you run docker without being root ? FINN needs docker to be launched without being root.
https://finn.readthedocs.io/en/latest/getting_started.html
https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user

**Reply by mustafatasci42** [reply]
Thanks for reply. Yes, I was able to run docker without root. I even added the following code to run_docker to confirm.
if [ "$FINN_DOCKER_RUN_AS_ROOT" = "0" ];then
  gecho "******* docker run as root *******"

![add_command_4_show](https://user-images.githubusercontent.com/15524481/186362348-d6eae3c3-1c79-41fc-8440-c26a112a26a4.png)
and when i run the code i see run as root message on the screen.

![run_as_root](https://user-images.githubusercontent.com/15524481/186362774-2c0c5a00-366c-411c-b094-8370f0ab82af.jpeg)

**Comment by julienmas** [comment]
Also you tried the suggested solution `pip install -U setuptools setuptools_scm wheel` ?

**Reply by mustafatasci42** [reply]
yes i installed it but the problem is not fixed.

**Comment by mustafatasci42** [comment]
Some dependencies (qonnx, brevitas and pyverilator) fail when run-docker.sh is run, while finn and finn_experimental work fine

**The docker_exec output is as follows**

`docker run 
-t 
--rm 
-it 
--tty 
--init 
--hostname finn_dev_mustafa 
-e SHELL=/bin/bash 
-w /home/mustafa/finn 
-v /home/mustafa/finn:/home/mustafa/finn 
-v /tmp/finn_dev_mustafa:/tmp/finn_dev_mustafa 
-e FINN_BUILD_DIR=/tmp/finn_dev_mustafa 
-e FINN_ROOT=/home/mustafa/finn 
-e LOCALHOST_URL=localhost 
-e VIVADO_IP_CACHE=/tmp/finn_dev_mustafa/vivado_ip_cache 
-e PYNQ_BOARD=Pynq-Z1 
-e PYNQ_IP= -e PYNQ_USERNAME=xilinx 
-e PYNQ_PASSWORD=xilinx 
-e PYNQ_TARGET_DIR=/home/xilinx/finn_dev_mustafa 
-e OHMYXILINX=/home/mustafa/finn/deps/oh-my-xilinx 
-e NUM_DEFAULT_WORKERS=4 
-v /etc/group:/etc/group:ro 
-v /etc/passwd:/etc/passwd:ro 
-v /etc/shadow:/etc/shadow:ro 
-v /etc/sudoers.d:/etc/sudoers.d:ro 
-v /home/mustafa/finn/ssh_keys:/home/mustafa/.ssh 
--user 1000:1000 

-v /tools/Xilinx:/tools/Xilinx 
-e XILINX_VIVADO=/tools/Xilinx/Vivado/2022.1 
-e VIVADO_PATH=/tools/Xilinx/Vivado/2022.1 
-e HLS_PATH=/tools/Xilinx/Vitis_HLS/2022.1 

xilinx/finn:v0.8.1-4-gabc50007-dirty.xrt_202210.2.13.466_18.04-amd64-xrt bash
`

**Comment by mustafatasci42** [comment, marked as answer]
The problem was due to the deps folder. I deleted the folder, ran run-docker.sh and it worked without errors with the newly created folder.

**Reply by cyrus-mz** [reply]
Would you please give some details/steps on how you did that?

I used git clone and am trying to install FINN on the home directory. There are similar issues such as "permission denied" for some directories like "qonnx" (which is fixable by changing the permission) and also this error:
 "LookupError: setuptools-scm was unable to detect version for '/home/cyrus/finn/deps/finn-experimental" for other ones like "finn-experimental".
 
 I do not have any idea how to fix it.

**Reply by mustafatasci42** [reply]
After installing Finn with git, delete the deps directory. Running run_docker.sh will recreate the deps directory. If the problem is not fixed, share the screenshot of the error, I will look again
